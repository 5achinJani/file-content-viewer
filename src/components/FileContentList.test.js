import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import {
  FileContentList,
  ContentLoader,
  ContentNotAvail,
  generateLineNo
} from "./FileContentList";

it("should render", () => {
  const props = {
    content: ["test"],
    page: 1,
    isLoading: false
  };
  const component = renderer.create(<FileContentList {...props} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("should generate correct line number based on page number and current line of content", () => {
  let params = {
    page: 1,
    index: 3
  };
  let lineNo = generateLineNo(params);
  expect(lineNo).toBe("04");

  params = {
    page: 1,
    index: 9
  };
  lineNo = generateLineNo(params);
  expect(lineNo).toBe("10");

  params = {
    page: 2,
    index: 5
  };
  lineNo = generateLineNo(params);
  expect(lineNo).toBe("16");

  params = {
    page: 11,
    index: 0
  };
  lineNo = generateLineNo(params);
  expect(lineNo).toBe("101");
});

it("should show loading component when loading", () => {
  const props = {
    content: [],
    page: 1,
    isLoading: true
  };
  const component = shallow(<FileContentList {...props} />);
  expect(component.equals(<ContentLoader />)).toBe(true);
});

it("should show empty content placeholder message when content is empty", () => {
  const props = {
    content: [],
    page: 1,
    isLoading: false
  };
  const component = shallow(<FileContentList {...props} />);
  expect(component.equals(<ContentNotAvail />)).toBe(true);
});
