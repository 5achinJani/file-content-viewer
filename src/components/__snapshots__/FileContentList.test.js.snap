// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th />
      </tr>
    </thead>
    <tbody
      name="js-content-list"
    >
      <tr>
        <td>
          01
          :
        </td>
        <td>
          test
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
